cluster-essentials/clusterinit.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2023 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package clusteressentials has APIs for bootstrapping kubernetes cluster
+package clusteressentials
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/cppforlife/go-cli-ui/ui"
+	"k8s.io/client-go/rest"
+
+	"github.com/vmware-tanzu/carvel-imgpkg/pkg/imgpkg/cmd"
+	"github.com/vmware-tanzu/tanzu-framework/util/carvelhelpers"
+	"github.com/vmware-tanzu/tanzu-framework/util/kube"
+)
+
+var (
+	downloadPackage = downloadClusterEssentialBundle
+)
+
+const tmpDir = "tmp"
+
+func downloadClusterEssentialBundle(clusterEssentialBundlePath string) error {
```

Comment:
If downloading imgpkg bundles is also needed in operations like upgrade, it'd be good to create this as a utility function.

Reply:
+1